### PR TITLE
Build: Don't run ESLint task in old Node

### PR DESCRIPTION
**Pull request by trentmwillis**
Addressing #1077.

### CODE REVIEW

**Comment by platinumazure** [comment]
Thanks a lot @trentmwillis for putting this together. The changes look good.

I do have one question, though. Wouldn't it make more sense for `grunt-eslint` to handle this? After all, they're a Grunt integration with ESLint and should be most aware of both sides. Shouldn't grunt-eslint detect Node version and/or ESLint version and export an empty task (possibly behind a silent failure option)? I feel like that would be the ideal way to handle this.

**Comment by trentmwillis** [comment]
> Wouldn't it make more sense for `grunt-eslint` to handle this?

Not really. They actually bumped a major version (`18` -> `19`) when they moved to ESLint 3 (which is when Node 0.12 was dropped I believe).  So from that perspective, they shouldn't need to support consumers using things outside of semver ranges (IMO). Additionally, this gives us an easy way to no-op any other potential tasks that might lose 0.12 support prior to us droppping 0.12 (which should be beginning of next year).

I could alternatively, try using a version of grunt-eslint with ESLint 2 instead of adding this additional infrastructure.

**Comment by platinumazure** [comment]
It just seemed like a lot of hoops to jump through in Grunt. I'm opposed to
using ESLint 2 because there have been many useful improvements in the 3.x
series, so if you believe the infrastructure you've created is the most
strategically sensible approach, I'll defer to your judgement.

**Comment by trentmwillis** [comment]
Yes, definitely more hoops to jump through, but from a maintainer perspective, I wouldn't want the grunt lib to add code to support a "non-use" case.

cc/ @leobalter @mgol if this looks good to y'all, I'll merge

File: package.json [hunk 1]
```diff
@@ -52,9 +52,8 @@
   },
   "scripts": {
     "browserstack": "sh build/run-browserstack.sh",
-    "ci": "grunt test",
-    "ci-include-lint": "grunt eslint && grunt test",
-    "test": "grunt",
+    "build": "grunt build",
+    "test": "grunt test",
```
Comment (leobalter) [inline package.json]: https://github.com/qunitjs/qunit/blob/master/test/index.html#L7

We still need to build the dist files for the tests, right?

Comment (leobalter) [inline package.json]: and now I look at the previous ci script and ... I'm missing something here.

Comment (trentmwillis) [inline package.json]: When `npm install` happens it runs the `prepublish` script which runs `grunt build`.

Comment (mgol) [inline package.json]: We shouldn't be using the `prepublish` hook at all, it's deprecated.

Comment (mgol) [inline package.json]: jQuery doesn't build on `grunt test` but it does on `npm test`:
https://github.com/jquery/jquery/blob/master/package.json#L63

I think it makes sense as it allows for the standard `npm install && npm test` (or `npm it` for short) workflow without diving into the project internals and without relying on a deprecated hook.

Comment (trentmwillis) [inline package.json]: Did not realize it was deprecated. Looks like it happened in npm 4. We could change it to the new `npm prepare` hook, but I think just including the build in the `test` command makes the most sense. Will update here shortly.

**Comment by trentmwillis** [comment]
@mgol updated to remove the prepublish script. @leobalter do you know if we were relying on this for releases or e browserstack tests? Browserstack in particular failed 100%, but I am unsure if it is because we don't generate a build now.

**Comment by leobalter** [comment]
It seems we need to add the build to the npm's `browserstack` script.

**Comment by leobalter** [comment]
LGTM

**Comment by leobalter** [comment]
Merged with the following commits:

- 172aa6366705d6a0a6f4497c557fd055574c4576
- 990c6985d1d27bfb348ab89847c751103cff256d
- da70f9d9e249897a6b213299e9c5c4dbbdc838de

**Pull request closed without merging**